gravitee-apim-console-webui/src/management/api/api-traffic-v4/analytics/api-analytics-message/api-analytics-message.component.spec.ts

```diff
@@ -374,6 +377,65 @@ describe('ApiAnalyticsMessageComponent', () => {
     });
   });
 
+  describe('Query parameters for enabled analytics', () => {
+    it('should display last day time range if no period query parameter found', async () => {
+      await initComponent();
+      expectAllAnalyticsCall();
+
+      const filtersBar = await componentHarness.getFiltersBarHarness();
+
+      const matSelect = await filtersBar.getMatSelect();
+      const selectedValue = await matSelect.getValueText();
+
+      expect(selectedValue).toEqual('Last day');
+    });
+
+    it('should display provided time range if period query parameter found', async () => {
+      await initComponent({ period: '1M' });
+      expectAllAnalyticsCall();
+
+      const filtersBar = await componentHarness.getFiltersBarHarness();
+
+      const matSelect = await filtersBar.getMatSelect();
+      const selectedValue = await matSelect.getValueText();
+
+      expect(selectedValue).toEqual('Last month');
+    });
+
+    it('should display default time range if period query parameter is incorrect', async () => {
+      await initComponent({ period: 'incorrect' });
+      expectAllAnalyticsCall();
+
+      const filtersBar = await componentHarness.getFiltersBarHarness();
+
+      const matSelect = await filtersBar.getMatSelect();
+      const selectedValue = await matSelect.getValueText();
+
+      expect(selectedValue).toEqual('Last day');
+    });
+
+    it('should display default time range if some other query parameter exist', async () => {
+      await initComponent({ otherParameter: 'otherParameter' });
+      expectAllAnalyticsCall();
+
+      const filtersBar = await componentHarness.getFiltersBarHarness();
+
+      const matSelect = await filtersBar.getMatSelect();
+      const selectedValue = await matSelect.getValueText();
+
+      expect(selectedValue).toEqual('Last day');
+    });
```

Comment:
This tests are pretty same it can be useful to write something like

```suggestion
    [
      {input: null, expected: 'Last day'},
      {input: { period: '1M' }, expected: 'Last month'},
      {input: { period: 'incorrect' }, expected: 'Last day'},
      {input: { otherParameter: 'otherParameter' }, expected: 'Last day'},
    ].forEach((testParams) => {
      it(`should display "${testParams.expected}" time range if query parameter is ${testParams.input}`, async () => {
        await initComponent(testParams.input);
        expectAllAnalyticsCall();
  
        const filtersBar = await componentHarness.getFiltersBarHarness();
  
        const matSelect = await filtersBar.getMatSelect();
        const selectedValue = await matSelect.getValueText();
  
        expect(selectedValue).toEqual(testParams.expected);
      });
    });
```

Reply:
Yeah, good point, thanks! I changed that for both test classes